### PR TITLE
go: add exercise B

### PR DESCRIPTION
**Pull request by ErikSchierboom**

### CODE REVIEW

File: exercises/go-2-b/.meta/start_3.go [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+		return 10
+	case 'X':
+		return 10
+	}
+
+	return int(card - '0')
+}
+
+func HandScore(hand string) int {
+	score := 0
```
Comment (tehsphinx) [inline exercises/go-2-b/.meta/start_3.go]:
```suggestion
	var score int
```

File: exercises/go-2-b/.meta/start_3.go [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
```
Comment (tehsphinx) [inline exercises/go-2-b/.meta/start_3.go]:
```suggestion
```

File: exercises/go-2-b/.meta/start_3.go [hunk 3]
```diff
@@ -0,0 +1,36 @@
+package go2b
+
+const MaximumHandScore = 21
+
+func CardValue(card rune) int {
```
Comment (tehsphinx) [inline exercises/go-2-b/.meta/start_3.go]: Should this be an internal function?

File: exercises/go-1-b/.meta/example_struct.go [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+	}
+
+	return score
+}
+
+func PlayerWins(playerHand string, dealerHand string) bool {
+	return HandScore(playerHand) > HandScore(dealerHand) &&
+		HandScore(playerHand) <= MaximumHandScore ||
+		HandScore(dealerHand) > MaximumHandScore
+}
```
[the remark this answers is not in the document]
Comment (ErikSchierboom) [reply exercises/go-1-b/.meta/example_struct.go]: For now, I've added your suggestion without precalculating the score. Is that okay?

**Comment by ErikSchierboom** [comment]
@tehsphinx Thanks for the review! I've updated the code.

File: exercises/go-1-b/.meta/example_struct.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (tehsphinx) [reply exercises/go-1-b/.meta/example_struct.go]: Sure, that is ok. 👍

File: exercises/go-1-b/.meta/example_struct.go [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+	Eight      = 8
+	Seven      = 7
+	Six        = 6
+	Five       = 5
+	Four       = 4
+	Three      = 3
+	Two        = 2
+)
+
+type Hand struct {
```
Comment (tehsphinx) [inline exercises/go-1-b/.meta/example_struct.go]: Wondering why this is exported. Everything else that would make this valueable to be exported is not exported: `score` method and `parseHand`.

File: exercises/go-1-b/.meta/example_struct.go [hunk 6]
```diff
@@ -0,0 +1,83 @@
+package go1b
+
+const maxScore = 21
+
+type Card int
```
Comment (tehsphinx) [inline exercises/go-1-b/.meta/example_struct.go]: Why export this? Same with all the constants. They are not usable in any valueable context.

File: exercises/go-2-b/.meta/start_2_example.go [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+	Ten        = 10
+	Nine       = 9
+	Eight      = 8
+	Seven      = 7
+	Six        = 6
+	Five       = 5
+	Four       = 4
+	Three      = 3
+	Two        = 2
+)
```
Comment (tehsphinx) [inline exercises/go-2-b/.meta/start_2_example.go]: Just a little trick that could be used here:

```go
const (
	Two Card = iota + 2
	Three
	Four
	Five
	Six
	Seven
	Eight
	Nine
	Ten
	Ace

	King  = 10
	Queen = 10
	Jack  = 10
)
```

No need to change this though if you'd rather have it the straight forward way.

Comment (tehsphinx) [reply exercises/go-2-b/.meta/start_2_example.go]: Also note, that in your code above, only `Ace` is of type `Card`. The others are of type `int`.

Comment (ErikSchierboom) [reply exercises/go-2-b/.meta/start_2_example.go]: Thanks!

**Comment by ErikSchierboom** [comment]
@tehsphinx The PR has been updated again according to your comments. Thanks a lot for the review!

**Comment by tehsphinx** [comment]
Looks good from my side :)

**Comment by ErikSchierboom** [comment]
Thanks @tehsphinx !

**Pull request merged**